doc/RELEASE.adoc

```diff
@@ -0,0 +1,120 @@
+Ad-hoc releases can be made upon request.
+
+We only make patch releases for version `x.y`, if a cardano-node release candidate depends on `x.y.z`, and an issue is found during the node release process that requires patching plutus.
+In all other cases, we always start a new major or minor release from master.
+
+== Release Process
+
+=== Major and Minor Releases
+
+Suppose we are releasing version `x.y.0`.
+
+1. Update version numbers in cabal files to `x.y.0` (if it is not already the case).
+- Version bounds for packages in the same repo should be set to `==x.y.*`.
+2. Make sure the change log is up to date.
+3. Tag `x.y.0-rc1` on master.
+4. Run all tests and checks.
+- This includes all tests that can be done with only the Plutus repo, such as unit tests, property-based tests, conformance tests and nightly tests.
+- This does not include tests that require building a new node, or integrating with other downstream projects.
+- In some cases we need input from domain experts and/or security audit (e.g., when adding a cryptography builtin). The right group of people should be involved and sign off on the tests and checks.
+5. If no release blocking issue is found, do a formal release and upload the packages to Hackage.
+- If issues are found, create a release branch `release/x.y`, fix the issues on master, backport the fixes to `release/x.y`, tag `x.y.0-rc2`, and go to step 4.
+- Why not just fix the issues on master and tag `x.y.0-rc2` from master?
+It is desirable to minimize the amount of change between `rc1` and `rc2`, because it may reduce the tests and checks that need to be performed against `rc2`.
+For instance, if `plutus-ledger-api` is the only package changed, there is no need to re-run tests on `plutus-core` or `plutus-tx`.
+Another example is if a security audit is done on `rc1`, and the changes in `rc2` does not modify the audited code, then the audit does not need to be re-done.
+
+6. After the release, update version numbers in cabal files to `x.y+1.0`.
+- This may not be the actual version of the next release - it could be `x+1.0.0`. We don't need to be precise here; we just need a version number higher than the latest release.
+
+=== Patch Releases
+
+1. If it is the first time we are making a patch release for version `x.y` (i.e., `x.y.0` -> `x.y.1`), a release branch may not exist for `x.y`. If so, create branch `release/x.y` from the `x.y.0` tag.
+- We create release branches lazily, because we do not expect to make many patch releases.
+2. Backport the needed fixes from master to `release/x.y`.
+3. Run all tests and checks.
+4. If no release blocking issue is found, do a formal release and upload the packages to Hackage.
+- If issues are found, fix them on master, backport the fixes to `release/x.y`, and go to step 3.
+
+== Backwards Compatibility
+
+Plutus releases should meet two backwards compatibility criteria:
+
+1. *Each Plutus release should build with the latest version of Cardano API.*
```

Comment:
`cardano-api` doesn't depend on `plutus` directly (or shouldn't). Do you mean the _ledger_ should build with the released version?

Reply:
Or no, I think you do mean this: you mean that _everything downstream_ should build, all the way to `cardano-api`.

Reply:
We need to automate this check! Probably not too hard with CHaP.

Reply:
Do we want this policy even for new _major_ versions? I buy it for minor versions, but kind of the point of major versions is that you break something, no?

Reply:
Yeah.. and minor versions shouldn't need this policy, because it's not supposed to be backwards compatible. So perhaps we don't need this policy at all!

Reply:
I rewrote this section in https://github.com/input-output-hk/plutus/pull/4885/commits/1c9f6a0431b35ab304ebddddd7e52d0aa4e5e645.